### ISSUE TITLE
Duplicate link reference definitions are not detected

### ISSUE THREAD

**Issue by dseynhae** [DavidAnson/markdownlint#425]
Consider:

```markdown
[test1][1]
![test2][2]
![test3][3]
[1]: https://example.com
[1]: https://commonmark.org/help/images/favicon.png
[4]: ./images/favicon.png
```

This will silently lead to problems in the rendered markdown

The correct code of course would be:
```markdown
[test1][1]
![test2][2]
![test3][3]

[1]: https://example.com
[2]: https://commonmark.org/help/images/favicon.png
[3]: ./images/favicon.png
```

Although linter can not predict which link should be hooked up with which definition, discrepancies should be flagged:
* Defined but unused references
* Used but undefined references
* Duplicate reference definitions
